test/server.cpp

```diff
@@ -1522,6 +1718,38 @@ TEST_F(ServerTest, CacheControlOfDynamicContent)
       EXPECT_TRUE(g->has_header("ETag")) << res;
     }
   }
+
+  // no_js_default
+  auto r = zfs1_->GET("/ROOT%23%3F/nojs");  
+  EXPECT_EQ(r->status, 200);
+  EXPECT_EQ(r->body,
+            HTML_PREAMBLE
+            FILTERS_HTML("")
+            HOME_BODY_TEXT("4")
+            POOR_ZIM_BOOK_HTML
+            TEST_ZIM_BOOK_HTML
+            RAY_CHARLES_BOOK_HTML
+            ZIM_CORNER_CASES_BOOK_HTML
+            FINAL_HTML_TEXT);
+
+  // no_js_eng_lang
+  r = zfs1_->GET("/ROOT%23%3F/nojs?lang=eng");
+  EXPECT_EQ(r->status, 200);
+  EXPECT_EQ(r->body,
+            HTML_PREAMBLE
+            FILTERS_HTML(" selected ")
+            HOME_BODY_TEXT("1")
+            RAY_CHARLES_BOOK_HTML
+            FINAL_HTML_TEXT);
+
+  // no_js_no_books
+  r = zfs1_->GET("/ROOT%23%3F/nojs?lang=fas");
+  EXPECT_EQ(r->status, 200);
+  EXPECT_EQ(r->body,
+            HTML_PREAMBLE
+            FILTERS_HTML("")
+            HOME_BODY_0_RESULTS
+            FINAL_HTML_TEXT);
 }
```

Comment:
I created a new file `test/html_dumper.cpp` for nojs tests (though, not too sure if that name is ok)

> In https://github.com/kiwix/libkiwix/pull/897#pullrequestreview-1334738349 I meant that the ServerTest.CacheControlOfDynamicContent unit-test must be enhanced to cover this new /nojs endpoint too.

I don't know how to do this, but added /nojs in `resources200Compressible`. If that's what was needed, cool! otherwise some guidance please! :>

Reply:
> I created a new file `test/html_dumper.cpp` for nojs tests (though, not too sure if that name is ok)

No, that's an overshoot. A new test-case in `test/server.cpp` is fine (but it must be placed reasonably).

> 
> > In [#897 (review)](https://github.com/kiwix/libkiwix/pull/897#pullrequestreview-1334738349) I meant that the ServerTest.CacheControlOfDynamicContent unit-test must be enhanced to cover this new /nojs endpoint too.
> 
> I don't know how to do this, but added /nojs in `resources200Compressible`. If that's what was needed, cool!

Cool! I wonder if ChatGPT would be able to do that (I hope that not).

Reply:
> No, that's an overshoot. A new test-case in test/server.cpp is fine (but it must be placed reasonably).

Okie removed and added them in a new test :> I hope this back and forth stuff hasn't annoyed you (yet!) :P

> Cool! I wonder if ChatGPT would be able to do that (I hope that not).

Hoping the same :P
But also waiting for the day when I tell GPT4 to do welcome page UI from a [paper drawing like this](https://twitter.com/heykahn/status/1635752848398102530)